workflow/tools/util_in_docker/generate_song_payload.py

```diff
@@ -235,29 +235,22 @@ def get_workflow_data_files(yaml_data):
     filename_keys = []
     for read_group in yaml_data.get('readGroups'):
         for file in read_group.get('files'):
-            if "song://" in file.get('fileName'):
-                repository = "collaboratory"
-            elif "file://" in file.get('fileName'):
+            if "file://" in file.get('path'):
                 repository = "local"
-            else:
-                repository = None
-
-            if repository == "collaboratory":
-                bundle_id = str(file.get('fileName')).split('/')[3]
-                object_id = str(file.get('fileName')).split('/')[4]
-            else:
                 bundle_id = None
                 object_id = None
+            else:
+                _, _, repository, bundle_id, object_id = str(file.get('path')).split('/')
 
-            if not os.path.basename(file.get('fileName')) in filename_keys:
+            if not file.get('fileName') in filename_keys:
```

Comment:
this should also change to use 'path', ie, `file.get('fileName') in filepath_keys`. Again, `path` is safe for uniqueness.